pkg/sql/catalog/nstree/catalog.go

```diff
@@ -25,10 +25,11 @@ import (
 )
 
 // Catalog is used to store an in-memory copy of the whole catalog, or a portion
-// thereof.
+// thereof, as well as metadata like comment and zone configs.
 type Catalog struct {
 	underlying NameMap
 	byteSize   int64
+	comments   map[keys.CommentKey]string
```

Comment:
Best use an ordered map like a b-tree here instead of a hash-map, in practice ordering does matter for methods like  `ForEachComment`.

Reply:
Good point. I think ordering doesn't matter actually since we really just need read/write comment by key, and we only iterate through all comments and upsert to another map... Let me see if I can name the `ForEachComment` thing to something else, maybe with something like "unordered" or what not. Oh my vocabulary...

Reply:
Alternatively you can build a slice in `ForEachComment`, sort it & iterate on it.